Requirements.md

```diff
@@ -0,0 +1,26 @@
+## Requirements
+
+1. Lazy initialization.
+   - DBConfigurations.cs line: 10.  
+2. Generics (in delegates, events and methods)
+    - All ```Base``` classes and interfaces. 
+3. Delegates.
+   - RestaurantService.cs line: 99
+   - FoodService.cs line: 95
+4. Events and their usage: standard and custom.
+   - ```RestaurantService``` publishes ```RestaurantRegistered``` event. ```EmailService``` subscribes to this event. When event is raised, email about successful registration is sent to restaurant's email adress.
+5. Exceptions and dealing with them in a meaningfull way.
+   - todo??? (Custom exceptions)
+6. Anonymous methods.
+   - EmailService.cs line: 15-19.
+7. Lambda expressions.
+   - EntitySorter.cs
+8. Concurrent programming (threading or async/await (for your own written classes); common resource usage between threads).
+   - EmailService.cs sending emails asyncronously.
+   - todo more???
```

Comment:
If we just use emailService, probably the teacher will ask us about tasks and that stuff. We are using databases on our next iteration, so we could try to migrate to databases and avoid refactor the code that we are deleting and study or write the possible questions that he could ask us

Reply:
We should account during the next practice lecture. I don't know if we will be able to migrate to EF and db until then. At least I won't be able to do that much work during the weekend.